app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestorePresenter.kt

```diff
@@ -1,34 +1,35 @@
 package org.p2p.wallet.auth.ui.restore.common
 
-import org.p2p.wallet.R
-import kotlinx.coroutines.launch
-import org.p2p.wallet.auth.interactor.UserSignUpInteractor
+import org.p2p.wallet.auth.interactor.OnboardingInteractor
+import org.p2p.wallet.auth.interactor.restore.SocialShareRestoreInteractor
+import org.p2p.wallet.auth.interactor.restore.UserRestoreInteractor
+import org.p2p.wallet.auth.repository.RestoreFlowDataLocalRepository
 import org.p2p.wallet.common.mvp.BasePresenter
-import timber.log.Timber
 
 class CommonRestorePresenter(
-    private val userSignUpInteractor: UserSignUpInteractor,
+    private val onboardingInteractor: OnboardingInteractor,
+    private val socialShareRestoreInteractor: SocialShareRestoreInteractor,
+    private val restoreUserRestoreInteractor: UserRestoreInteractor,
+    private val restoreFlowDataLocalRepository: RestoreFlowDataLocalRepository
 ) : BasePresenter<CommonRestoreContract.View>(), CommonRestoreContract.Presenter {
 
     override fun useGoogleAccount() {
         view?.startGoogleFlow()
     }
 
-    override fun setAlternativeIdToken(userId: String, idToken: String) {
-        launch {
-            // TODO modify logic with userSignInInteractor
-            when (val result = userSignUpInteractor.trySignUpNewUser(idToken, userId)) {
-                is UserSignUpInteractor.SignUpResult.SignUpSuccessful -> {
-                    view?.onSuccessfulSignUp()
-                }
-                is UserSignUpInteractor.SignUpResult.SignUpFailed -> {
-                    Timber.e(result, "Restoring account failed")
-                    view?.showErrorSnackBar(R.string.error_general_message)
-                }
-                UserSignUpInteractor.SignUpResult.UserAlreadyExists -> {
-                    view?.onNoTokenFoundError(userId)
-                }
-            }
+    override fun switchFlowToRestore() {
+        onboardingInteractor.currentFlow = OnboardingInteractor.OnboardingFlow.RESTORE_WALLET
+        restoreFlowDataLocalRepository.generateRestoreUserKeyPair()
+    }
+
+    override fun setGoogleIdToken(userId: String, idToken: String) {
+        view?.setLoadingState(isScreenLoading = true)
+        socialShareRestoreInteractor.restoreSocialShare(idToken, userId)
+        view?.setLoadingState(isScreenLoading = false)
+        if (restoreUserRestoreInteractor.isUserReadyToBeRestored()) {
+            // navigate to pin if social + device share exists
```

Comment:
the navigation will be done in another ticket?

Reply:
yes, this is navigation when we have a device+social share
i can put a task number here